ygen/protogen.go

```diff
@@ -358,6 +387,37 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
 	return append(msgDefs, msgDef), errs
```

Comment:
Per comments in other review, this function is too long and has too much nesting.

Reply:
Split out leaves in this version too. PTAL.